streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -31,7 +31,9 @@
  *     WHERE
  *       stream1.key = stream2.key
  *       AND
- *       stream2.ts - before <= stream1.ts <= stream2.ts + after
+ *       stream2.ts >= stream1.ts - before
+ *       AND
+ *       stream2.ts <= stream1.ts + after
```

Comment:
Ok. (BTW: `BETWEEN` is defined to be an closed interval in SQL, so it would be actually be well defined :) )